scripts/verify-test-resource.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+set -o nounset # don't allow use of undefined variables
+set -o pipefail # if any single command in a pipe chain exits non-zero, set the exit status to that
+
+# Set environment variable for GitHub action to recognise colours
+export TERM=xterm
+
+# Colours for stdout
+RED=$(tput setaf 1)
+GREEN=$(tput setaf 2)
+WHITE=$(tput setaf 7)
+NC=$(tput sgr0)
+RED_BG=$(tput setab 1)
+
+# Path to test resources
+TEST_RESOURCES_PATH=src/test/resources
+
+# Temporary directory for saving compiled classes
+TEMP_DIR_COMPILED_RESOURCES=$(mktemp -d)
+echo "Created $TEMP_DIR_COMPILED_RESOURCES for saving compiled files temporarily."
+echo -e "It will be removed after the script has executed with any exit code.\n"
+
+# Delete the temporary directory as it served its purpose
+cleanup() {
+  echo -e "\nRemoving $TEMP_DIR_COMPILED_RESOURCES directory ..."
+  rm -rf "$TEMP_DIR_COMPILED_RESOURCES"
+  echo "Done."
+}
+
+# Make sure the temporary directory is deleted even if the script aborts
+trap cleanup EXIT
+
+# Global variables which will be modified as the script runs
+status=0
+file_count=0
+compilation_errors=0
+
+# Verify if the resources compiles
+compile() {
+  local file=$1 # path to program in /tmp or equal to actual path
+  local actual_path=$2 # actual path of the test resource
+  if javac "$file" -d "$TEMP_DIR_COMPILED_RESOURCES"; then
+    echo -e "${actual_path}:${GREEN} Compiled successfully!${NC}"
+  else
+    echo -e "${RED}There were compilation errors in ${WHITE}${RED_BG}${actual_path}${NC}"
+    status=1
+    compilation_errors=$((compilation_errors+1))
+  fi
+}
+
+# Rename file according to the first public class inside
+get_and_rename_file() {
```

Comment:
This function checks for the **first** public class inside the java file and copies the file to `/tmp` directory with the name of the public class. Let me know if this is the best way or not.